eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/Handler.php

```diff
@@ -93,33 +93,49 @@ public function __construct(
         $this->slugConverter = $slugConverter;
     }
 
+    public function publishUrlAliasForLocation(
+        $locationId,
+        $parentLocationId,
+        $name,
+        $languageCode,
+        $alwaysAvailable = false,
+        $updatePathIdentificationString = false
+    ) {
+        $languageId = $this->languageHandler->loadByLanguageCode($languageCode)->id;
+
+        $this->internalPublishUrlAliasForLocation(
+            $locationId,
+            $parentLocationId,
+            $name,
+            $languageId,
+            $alwaysAvailable,
+            $updatePathIdentificationString
+        );
+    }
+
     /**
-     * This method creates or updates an urlalias from a new or changed content name in a language
-     * (if published). It also can be used to create an alias for a new location of content.
-     * On update the old alias is linked to the new one (i.e. a history alias is generated).
-     *
-     * $alwaysAvailable controls whether the url alias is accessible in all
-     * languages.
+     * todo document.
```

Comment:
Added in https://github.com/ezsystems/ezpublish-kernel/pull/1781/commits/834e2215528bc858031bf0e5f0a8dba8a7d3a704.